src/Tools/dotnet-monitor/CollectionRules/Configuration/CustomShortcutsConfigureNamedOptions.cs

```diff
@@ -0,0 +1,91 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Microsoft.Diagnostics.Tools.Monitor.CollectionRules.Actions;
+using Microsoft.Diagnostics.Tools.Monitor.CollectionRules.Options;
+using Microsoft.Diagnostics.Tools.Monitor.CollectionRules.Triggers;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.Options;
+
+namespace Microsoft.Diagnostics.Tools.Monitor.CollectionRules.Configuration
+{
+    internal sealed class CustomShortcutsConfigureOptions :
+        IConfigureOptions<CustomShortcutOptions>
+    {
+        private readonly ICollectionRuleTriggerOperations _triggerOperations;
+        private readonly ICollectionRuleActionOperations _actionOperations;
+        private readonly IConfiguration _configuration;
+
+        public CustomShortcutsConfigureOptions(
+            ICollectionRuleTriggerOperations triggerOperations,
+            ICollectionRuleActionOperations actionOperations,
+            IConfiguration configuration)
+        {
+            _triggerOperations = triggerOperations;
+            _actionOperations = actionOperations;
+            _configuration = configuration;
+        }
+
+        public void Configure(CustomShortcutOptions options)
+        {
+            // Action Shortcuts
+            foreach (var key in options.Actions.Keys)
+            {
+                IConfigurationSection section = _configuration.GetSection(ConfigurationPath.Combine(nameof(RootOptions.CustomShortcuts), nameof(CustomShortcutOptions.Actions), key));
+
+                if (section.Exists())
+                {
+                    section.Bind(options);
```

Comment:
`options` is of type `CustomShortcutOptions`. Here, you are binding this options instance to the "CustomShortcuts:Actions:<key>" section, which I expect to be representative of a `CollectionRuleActionOptions` instance. As is, I think this code is not doing any under normal circumstances; under misconfigured circumstances, I think it will lead to unexpected behavior.